### ISSUE TITLE
[Bug?]: http status code and headers not set when deployed

### ISSUE THREAD

**Issue by ryoid** [solidjs/solid-start#1678]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

Setting http status code and header works locally as expected. However, when deployed on Cloudflare the response is 200, without the custom header.

https://developers.cloudflare.com/pages/framework-guides/deploy-a-solid-site/

```tsx
// [...404].tsx
import { HttpHeader, HttpStatusCode } from "@solidjs/start";

export default function NotFound() {
  return (
    <main>
      <HttpStatusCode code={404} />
      <HttpHeader name="x-custom-header" value="test" />
    </main>
  );
}
```

 Might be related to https://github.com/solidjs/solid-start/issues/1643

### Expected behavior 🤔

Receive the status 404 and custom header when visiting a 404 page.

### Steps to reproduce 🕹

Steps:

https://developers.cloudflare.com/pages/framework-guides/deploy-a-solid-site/

1. Create a SolidStart project with the bare template
2. Add a `<HttpHeader />` to the 404 page
3. Deploy on Cloudflare

### Context 🔦

Suffering from build assets 404 pages being cached.

**Comment by NawfelBgh** [comment]
I confirm that `HttpHeader` doesn't works correctly, but both in DEV and BUILT mode (in version "2.0.0-alpha.2").

I created a reproducing repo with test results in https://github.com/NawfelBgh/solid-start-test-http-header

## On DEV mode

```sh
npm run dev
```

- Just after starting the server, the `cache-control` response header **is missing**:

```sh
curl -I http://localhost:5173/about
HTTP/1.1 200 OK
Vary: Origin
content-type: text/html
Date: Thu, 07 May 2026 23:50:29 GMT
Connection: keep-alive
Keep-Alive: timeout=5
```

- Starting from the second request onward, we do receive the header:

```sh
curl -I http://localhost:5173/about
HTTP/1.1 200 OK
Vary: Origin
cache-control: public, max-age=600
content-type: text/html
Date: Thu, 07 May 2026 23:50:32 GMT
Connection: keep-alive
Keep-Alive: timeout=5
```

## On BUILT mode

```sh
npm run build
npm run preview

curl -I http://localhost:4173/about       
```

- Just after starting the server, the `cache-control` response header **is missing**
- Starting from the second request onward, we do receive the header

(Just like in dev mode).

Bizarrely, curl is not showing a worse problem which appears in built mode, both in Chrome and in Firefox:

- To the first request to the about page we get the response headers:

```sh
HTTP/1.1 200 OK
0: content-type
1: text/html
Vary: Origin
content-encoding: identity
Date: Fri, 08 May 2026 00:24:49 GMT
Connection: keep-alive
Keep-Alive: timeout=5
Transfer-Encoding: chunked
```

Notice that the `content-type` header is improperly encoded, in addition to `cache-control` being missing.

- Starting from the second request onward, we do receive the header

```sh
HTTP/1.1 200 OK
0: cache-control
1: public, max-age=600
2: content-type
3: text/html
Vary: Origin
content-encoding: identity
Date: Fri, 08 May 2026 00:26:23 GMT
Connection: keep-alive
Keep-Alive: timeout=5
Transfer-Encoding: chunked
```

Notice that both `cache-control` and `content-type` headers are improperly encoded. Because of this the page is not cached by the browser.

**Comment by birkskyum** [comment]
Can you update this to teh RC release of SolidStart v2, and use nitro v3? THe nitro v2 vite pluign is deprecated

**Comment by birkskyum** [comment]
Fixed in `2.0.0-rc.3` by #2231, which defers the response until the stream shell is ready. Retested your snippet on v2 with Nitro v3 in `vite dev`, node preview, and the `cloudflare_module` preset under `wrangler dev`: 404 status and both headers come through in all three, including on the first request. The deferred-async case (header inside a `<Show>` gated on `createAsync({ deferStream: true })`) works too, and the malformed `0: content-type` encoding reported above is gone.

Please upgrade to `2.0.0-rc.3` and reopen if you still see it.
